air/src/air/coefficients.rs

```diff
@@ -73,16 +73,38 @@ pub struct LagrangeConstraintsCompositionCoefficients<E: FieldElement> {
 ///
 /// 1. The evaluation proofs for each trace polynomial at $z$ and $g \cdot z$ can be batched using
 /// the non-normalized Lagrange kernel over the set $\{z, g \cdot z\}$. This, however, requires
-/// that the FRI protocol is run with rate $\rho^{+} := \frac{\kappa + 2}{\nu}$ where $\kappa$ and
-/// $\nu$ are the length of the execution trace and the LDE domain size, respectively.
+/// that the FRI protocol is run with a larger agreement parameter
+/// $\alpha^{+} = (1 + 1/2m)\cdot\sqrt{\rho^{+}}$ where $\rho^{+} := \frac{\kappa + 2}{\nu}$,
+/// $\kappa$ and $\nu$ are the length of the execution trace and the LDE domain size, respectively.
 /// 2. The resulting $Y(x)$ do not need to be degree adjusted but the soundness error of the
 /// protocol needs to be updated. For most combinations of batching parameters, this leads to a
 /// negligible increase in soundness error. The formula for the updated error can be found in
 /// Theorem 8 of https://eprint.iacr.org/2022/1216.
+///
+/// In the case when the trace polynomials contain a trace polynomial corresponding to a Lagrange
+/// kernel column, the above expression of $Y(x)$ includes the additional term given by
+///
+/// $$
+/// \gamma \cdot \frac{T_l(x) - p_S(x)}{Z_S(x)}
+/// $$
```

Comment:
Is this equivalent to:

$$
\gamma \cdot \sum_{i=0}^{log(n)}\frac{T_l(x) - T_l(z_i)}{x - z_i}
$$

Reply:
Yes, both forms achieve the same effect with the same soundness degradation. So originally we had always two (opening) points `z` and `g.z` but now, only for Lagrange kernel, we have $z, g^{2^0}\cdot z, \cdots, g^{2^{\log(n) -1}}\cdot z$ i.e. $\log(n) + 1$ (opening) points.